crates/iroha_core/src/sumeragi/main_loop.rs

```diff
@@ -841,57 +838,74 @@ impl Sumeragi {
             .expect("INTERNAL BUG: transactions in block exceed usize::MAX");
         let block_time = state.world.view().parameters.sumeragi.block_time();
         let tx_cache_full = self.transaction_cache.len() >= max_transactions.get();
-        let deadline_reached = self.round_start_time.elapsed() > block_time;
+        let deadline_reached =
+            self.topology.view_change_index() > 0 || self.round_start_time.elapsed() > block_time;
         let tx_cache_non_empty = !self.transaction_cache.is_empty();
 
         if tx_cache_full || (deadline_reached && tx_cache_non_empty) {
             let transactions = self
                 .transaction_cache
                 .iter()
-                .map(|tx| tx.deref().clone())
+                .map(|tx| CommittedTransaction {
+                    value: tx.deref().clone().into(),
+                    error: None,
+                })
```

Comment:
this should happen inside `BlockBuilder::new`. Restore it and move this code there